static/specification.html

```diff
@@ -194,51 +191,60 @@ <h3>
-                // See how much was received and in what currency
+                // See how much was sent and in what currency
                 const { amount, assetCode, assetScale } = event;
-                console.log(`Received ${assetCode}${amount} (${assetScale}).`);
+                console.log(`Browser sent ${assetCode}${amount / (10 * assetScale)}.`);
 
-                // Use the receipt to verify the payment using
-                // a verification service (see next example).
-                verifyReceipt(event);
+                // Use the incoming payment URL to verify the payment at the receiver.
+                verifyPayment(event);
               });
             &lt;/script&gt;
           </pre>
         </aside>
       </section>
       <section>
         <h3>
-          Verifying a receipt
+          Verifying a payment
         </h3>
         <p>
-          The {{MonetizationEvent}}'s {{MonetizationEvent/receipt}} attribute
-          is a unique string that can be used to verify the payment using a
-          payment verification service.
+          The {{MonetizationEvent}}'s {{MonetizationEvent/incomingPayment}}
+          attribute is a URL that can be used to verify the payment at the
+          [=monetization receiver=].
         </p>
         <p>
-          Below is a hypothetical verification service "verifier.example" that
-          performs verification via a `POST` HTTP request.
+          In most cases requests to the URL will need to be authenticated.
+          Below is a hypothetical verification endpoint that will make
+          authenticated requests to the receiver and return the results.
         </p>
         <aside class="example">
           <pre class="JS">
             /** @type {MonetizationEvent} event */
-            async function verifyReceipt(event) {
+            async function verifyPayment(event) {
+              // Legacy receivers don't support returning incoming payment URLs
+              if(!event.incomingPayment){
+                throw new Error('No incoming payment URL')
+              }
+
               const body = JSON.stringify({
-                receipt: event.receipt
+                incomingPayment: event.incomingPayment
               });
 
-              const response = await fetch('https://verifier.example/verify/', {
+              const response = await fetch('/verify', {
                 method: 'POST',
+                credentials: 'same-origin',
+                mode: 'same-origin',
+                cache: 'no-cache',
                 headers: {
                   'Content-Type': 'application/json'
                 },
                 body);
               });
 
               if (response.ok) {
-                // The payment was verified
-              } else {
-                // The payment was not verified
+                // The incoming payment was fetched successfully
+                const { receivedAmount } = JSON.parse(response.json())
```

Comment:
If I understood you correctly on slack, the incoming payments could be reused across page refreshes? (You said something about the Agent seeing the same payment pointer and reusing the incoming_payment)

Reply:
Yes. But that is outside the scope of this spec. That would be up to the UA to decide. What the website would get in the event is the URL of the incoming payment that is being paid into currently.

Reply:
It's quite a change to move from an `event/`amount` per packet to something where you get one event per payment initiation, with a received amount that will be the *current* received amount, and may include amounts from previous payment initiations. So I'm curious if there are implications that aren't immediately obvious. 

On the server side, how will the incoming_payment be used across sessions. How will it know when fresh payment has come and it's not looking at an amount for services/content already provided etc. Will it need to keep to some state somewhere?